arrow-core-data/src/main/kotlin/arrow/typeclasses/FunctorFilter.kt

```diff
@@ -16,19 +14,25 @@ interface FunctorFilter<F> : Functor<F> {
   /**
    * A combined map and filter. Filtering is handled via [Option] instead of [Boolean] such that the output type [B] can be different than the input type [A].
    */
+  @Deprecated("Please use mapNotNull")
   fun <A, B> Kind<F, A>.filterMap(f: (A) -> Option<B>): Kind<F, B>
 
+  fun <A, B> Kind<F, A>.mapNotNull(f: (A) -> B?): Kind<F, B> = filterMap { Option.fromNullable(f(it)) }
+
   /**
    * "Flatten" out a structure by collapsing Options.
    */
+  @Deprecated("Deprecated because Option is deprecated")
   fun <A> Kind<F, Option<A>>.flattenOption(): Kind<F, A> = filterMap(::identity)
 
+  fun <A> Kind<F, A?>.flattenNullable(): Kind<F, A> = mapNotNull(::identity)
```

Comment:
also this one... not sure if it makes sense :smile:

Reply:
I think this should be `filterNotNull` in line with https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/filter-not-null.html